cpp/src/arrow/compute/key_hash.h

```diff
@@ -219,5 +219,24 @@ class ARROW_EXPORT Hashing64 {
                       const uint8_t* keys, uint64_t* hashes);
 };
 
+template <typename T = uint32_t>
+static int64_t EstimateBatchStackSize(int32_t batch_size) {
```

Comment:
Do we need to export this?

Reply:
Actually, i want to unify the logic in Hashing32 and Hashing64. But seems unnecessary.